packages/cloud/src/attachment/withCloudAttachmentOverrides.ts

```diff
@@ -0,0 +1,75 @@
+import { EElementOrText, insertNode, Value } from '@udecode/plate-core';
+import Defer from 'p-defer';
+import { CloudEditor } from '../cloud/types';
+import { UploadError, UploadSuccess } from '../upload';
+import { TCloudAttachmentElement } from './types';
+
+type CloudAttachmentValue = TCloudAttachmentElement[];
+
+// const uploadMap = new Map<string, Atom<Upload>>();
+
+export function withCloudAttachmentOverrides<
+  V extends Value = Value,
+  E extends CloudEditor<V> = CloudEditor<V>
+>(editor: E) {
+  /**
+   * We create a deferredFinish which is an object with a `promise` and a way
+   * to `resolve` or `reject` the Promise outside of the Promise. We use
+   * `p-defer` library to do this. The `finish` Promise gets added to the
+   * `origin` object so we can await `origin.finish` during the save process
+   * to wait for all the files to finish uploading.
+   */
+  const deferredFinish = Defer<UploadSuccess | UploadError>();
+  const finishPromise = deferredFinish.promise;
+
+  editor.cloud.genericFileHandlers = {
+    onStart(e) {
+      console.log('start', e);
+      const node: EElementOrText<CloudAttachmentValue> = {
+        type: 'cloud_attachment',
```

Comment:
Use `getPluginType`

Reply:
Hi Ziad, I added `getPluginType` to this but just a heads up that the `plugin` argument wasn't being used which raised an eslint issue so I added an eslint-diable-next-line comment to get around that.